mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -0,0 +1,255 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Contains the wrapper class for SMAC Bayesian optimizers.
+See Also: <https://automl.github.io/SMAC3/main/index.html>
+"""
+
+from pathlib import Path
+from typing import Mapping, Optional
+from tempfile import TemporaryDirectory
+
+import ConfigSpace
+import numpy.typing as npt
+import pandas as pd
+
+from mlos_core.optimizers.bayesian_optimizers import BaseBayesianOptimizer
+from mlos_core.spaces.adapters.adapter import BaseSpaceAdapter
+
+
+class SmacOptimizer(BaseBayesianOptimizer):
+    """Wrapper class for SMAC based Bayesian optimization.
+
+    Parameters
+    ----------
+    parameter_space : ConfigSpace.ConfigurationSpace
+        The parameter space to optimize.
+
+    space_adapter : BaseSpaceAdapter
+        The space adapter class to employ for parameter space transformations.
+
+    seed : Optional[int]
+        By default SMAC uses a known seed (0) to keep results reproducible.
+        However, if a `None` seed is explicitly provided, we let a random seed be produced by SMAC.
+
+    run_name : Optional[str]
+        Name of this run. This is used to easily distinguish across different runs.
+        If set to `None` (default), SMAC will generate a hash from metadata.
+
+    output_directory : Optional[str]
+        The directory where SMAC output will saved. If set to `None` (default), a temporary dir will be used.
+
+    max_trials : int
+        Maximum number of trials (i.e., function evaluations) to be run. Defaults to 100.
+        Note that modifying this value directly affects the value of `n_random_init`, if latter is set to `None`.
+
+    n_random_init : Optional[int]
+        Number of points evaluated at start to bootstrap the optimizer. Defaults to 10.
+
+    n_random_probability: Optional[float]
+        Probability of choosing to evaluate a random configuration during optimization.
+        Defaults to `0.1`. Setting this to a higher value favors exploration over exploitation.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments,too-many-locals
+        self,
+        parameter_space: ConfigSpace.ConfigurationSpace,
+        space_adapter: Optional[BaseSpaceAdapter] = None,
+        seed: Optional[int] = 0,
+        run_name: Optional[str] = None,
+        output_directory: Optional[str] = None,
+        max_trials: int = 100,
+        n_random_init: Optional[int] = 10,
+        n_random_probability: Optional[float] = 0.1,
+    ):
+        super().__init__(parameter_space, space_adapter)
```

Comment:
we ues named parameters in the base constructor, too, so you'll need that:
```suggestion
        super().__init__(
            parameter_space=parameter_space,
            space_adapter=space_adapter,
        )
```